api/tests/integration/AuthTest.php

```diff
@@ -274,4 +278,167 @@ public function testMakeLoginToken()
 
         $this->assertEquals($id, $user->user_id);
     }
+
+    public function testEnvironmentForSocial()
+    {
+        // Make sure we have the hosts env variable for the redirect url generation
+        $hostApi = $this->app['config']['hosts.api'];
+        $hostApp = $this->app['config']['hosts.app'];
+
+        $this->assertNotNull($hostApi);
+        $this->assertStringStartsWith('http', $hostApi);
+        $this->assertStringEndsNotWith('/', $hostApi);
+        $this->assertNotNull($hostApp);
+        $this->assertStringStartsWith('http', $hostApp);
+        $this->assertStringEndsNotWith('/', $hostApp);
+    }
+
+    public function testInvalidProvider()
+    {
+        $this->get('/auth/social/foobar');
+
+        $this->assertException('provider', 404, 'NotFoundHttpException');
+    }
+
+    public function testProviderRedirect()
+    {
+        $this->get('/auth/social/facebook');
+
+        $this->assertResponseStatus(302);
+        $this->assertContains('refresh', $this->response->getContent());
+    }
+
+    public function testProviderRedirectReturnUrlOAuthOne()
+    {
+        $returnUrl = 'http://www.foo.bar/';
+
+        // If we have no valid twitter credentials, we'll mock the redirect
+        // request and set a cache manually. That allows us to still run live
+        // tests against twitter if credentials is available, and if not
+        // available, we still can test that the cache with the returnurl is
+        // properly set.
+        if (!$this->app->config->get('services.twitter.client_id') ||
+            !$this->app->config->get('services.twitter.client_id')
+        ) {
+            Cache::put('oauth_return_url_'.'foobar', $returnUrl, 1);
+            $mock = Mockery::mock('App\Extensions\Socialite\SocialiteManager');
+            $this->app->instance('Laravel\Socialite\Contracts\Factory', $mock);
+            $mock->shouldReceive('with->stateless->redirect')
+                ->once()
+                ->andReturn(redirect('http://foo.bar?oauth_token=foobar'));
+        }
+
+        $this->get('/auth/social/twitter?returnUrl='.urlencode($returnUrl));
+
+        // Parse the oauth token from the response and get the cached value
+        $segments = parse_url($this->response->headers->get('location'));
+        parse_str($segments['query'], $array);
```

Comment:
It is probably a good idea to assert that the `location` header is set as the exception thrown by `null['query']` is a pain to debug.